### PR TITLE
chore(chart-unfurls): Add a warning in Slack integration

### PR DESCRIPTION
**Pull request by shruthilayaj**
With discover charts unfurl in slack being EA'ed add a note
in the Slack integration for those who have the feature
letting them know that project permissions will not be
honoured with chart unfurls.

Screenshots without and with the feature flag:
Screen Shot 2021-09-02 at 7 21 46 AM
Screen Shot 2021-09-02 at 7 21 55 AM

### CODE REVIEW

**Review by wmak — APPROVED** [review]
Lgtm

File: src/sentry/integrations/slack/integration.py [hunk 1]
```diff
@@ -50,14 +50,21 @@
     "text": "The Slack integration adds a new Alert Rule action to all projects. To enable automatic notifications sent to Slack you must create a rule using the slack workspace action in your project settings.",
 }
 
+discover_unfurl_alert = {
+    "type": "warning",
+    "text": "Project permissions will not apply when unfurling Sentry Discover charts your Slack workspace.",
```
Comment (dashed) [inline src/sentry/integrations/slack/integration.py:55]:
```suggestion
    "text": "Project permissions will not apply when unfurling Sentry Discover charts from within your Slack workspace.",
```

Comment (shruthilayaj) [reply src/sentry/integrations/slack/integration.py:55]: oof good catch

**Review by dashed — APPROVED** [review]

**Review by manuzope — APPROVED** [review]
This is fine for EA but I know there are some orgs (the ones with closed membership) who care a lot about one team not having access to information for another team. Sounds like a product decision cc: @jonesphillip @jeremybourque

**Pull request merged**